Revert DynaLoader to 5.8.3 version so as not to break programs linked
with pre-5.8.4 libperl.so (and DynaLoader.a).

diff -ur perl-5.8.4/ext/DynaLoader/DynaLoader_pm.PL perl-5.8.3/ext/DynaLoader/DynaLoader_pm.PL
--- perl-5.8.4/ext/DynaLoader/DynaLoader_pm.PL	2004-04-04 23:31:25.000000000 +1000
+++ perl-5.8.3/ext/DynaLoader/DynaLoader_pm.PL	2003-12-13 23:40:30.000000000 +1100
@@ -29,7 +29,7 @@
 
 use vars qw($VERSION *AUTOLOAD);
 
-$VERSION = '1.05';	# avoid typo warning
+$VERSION = '1.04';	# avoid typo warning
 
 require AutoLoader;
 *AUTOLOAD = \&AutoLoader::AUTOLOAD;
@@ -80,7 +80,6 @@
 my $Mac_FS;
 $Mac_FS = eval { require Mac::FileSpec::Unixish } if $Is_MacOS;
 
-@dl_shared_objects  = ();       # shared objects for symbols we have 
 @dl_require_symbols = ();       # names of symbols we need
 @dl_resolve_using   = ();       # names of files to link with
 @dl_library_path    = ();       # path to look for files
@@ -340,9 +339,6 @@
     my $xs = dl_install_xsub("${module}::bootstrap", $boot_symbol_ref, $file);
 
     # See comment block above
-
-	push(@dl_shared_objects, $file); # record files loaded
-
     &$xs(@args);
 }
 
@@ -559,7 +555,6 @@
   $dl_debug
   @dl_librefs
   @dl_modules
-  @dl_shared_objects
                                                   Implemented in:
   bootstrap($modulename)                               Perl
   @filepaths = dl_findfile(@names)                     Perl
@@ -635,10 +630,6 @@
 
 An array of module (package) names that have been bootstrap'ed.
 
-=item @dl_shared_objects
-
-An array of file names for the shared objects that were loaded.
-
 =item dl_error()
 
 Syntax:
diff -ur perl-5.8.4/ext/DynaLoader/dl_hpux.xs perl-5.8.3/ext/DynaLoader/dl_hpux.xs
--- perl-5.8.4/ext/DynaLoader/dl_hpux.xs	2004-01-20 08:54:32.000000000 +1100
+++ perl-5.8.3/ext/DynaLoader/dl_hpux.xs	2001-10-25 12:34:02.000000000 +1000
@@ -102,19 +102,6 @@
         sv_setiv( ST(0), PTR2IV(obj) );
 
 
-int
-dl_unload_file(libref)
-    void *	libref
-  CODE:
-    DLDEBUG(1,PerlIO_printf(Perl_debug_log, "dl_unload_file(%lx):\n", PTR2ul(libref)));
-    RETVAL = (shl_unload(libref) == 0 ? 1 : 0);
-    if (!RETVAL)
-	SaveError(aTHX_ "%s", Strerror(errno));
-    DLDEBUG(2,PerlIO_printf(Perl_debug_log, " retval = %d\n", RETVAL));
-  OUTPUT:
-    RETVAL
-
-
 void *
 dl_find_symbol(libhandle, symbolname)
     void *	libhandle
